go/internal/controller/translator/agent/tls_mounting_test.go

```diff
@@ -0,0 +1,272 @@
+func Test_addTLSConfiguration_CustomCertKey(t *testing.T) {
+	mdd := &modelDeploymentData{}
+	tlsConfig := &v1alpha2.TLSConfig{
+		CACertSecretRef: "internal-ca-cert",
+		CACertSecretKey: "custom-ca.pem",
+	}
+
+	addTLSConfiguration(mdd, tlsConfig)
+
+	// Verify TLS_CA_CERT_PATH uses custom key
+	foundCACertPath := false
+	expectedPath := "/etc/ssl/certs/custom/custom-ca.pem"
+	for _, env := range mdd.EnvVars {
+		if env.Name == "TLS_CA_CERT_PATH" {
+			foundCACertPath = true
+			if env.Value != expectedPath {
+				t.Errorf("Expected TLS_CA_CERT_PATH=%s, got %s", expectedPath, env.Value)
+			}
+		}
+	}
+	if !foundCACertPath {
+		t.Error("TLS_CA_CERT_PATH environment variable not found")
+	}
+}
+
+// Test_addTLSConfiguration_UseSystemCAsFlag verifies UseSystemCAs env var
+func Test_addTLSConfiguration_UseSystemCAsFlag(t *testing.T) {
+	tests := []struct {
+		name         string
+		useSystemCAs bool
+		expected     string
+	}{
+		{
+			name:         "UseSystemCAs true",
+			useSystemCAs: true,
+			expected:     "true",
+		},
+		{
+			name:         "UseSystemCAs false",
+			useSystemCAs: false,
+			expected:     "false",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mdd := &modelDeploymentData{}
+			tlsConfig := &v1alpha2.TLSConfig{
+				UseSystemCAs: tt.useSystemCAs,
+			}
+
+			addTLSConfiguration(mdd, tlsConfig)
+
+			foundUseSystemCAs := false
+			for _, env := range mdd.EnvVars {
+				if env.Name == "TLS_USE_SYSTEM_CAS" {
+					foundUseSystemCAs = true
+					if env.Value != tt.expected {
+						t.Errorf("Expected TLS_USE_SYSTEM_CAS=%s, got %s", tt.expected, env.Value)
+					}
+				}
+			}
+			if !foundUseSystemCAs {
+				t.Error("TLS_USE_SYSTEM_CAS environment variable not found")
+			}
+		})
+	}
+}
+
+// Test_addTLSConfiguration_AllFieldsCombined verifies all fields work together
+func Test_addTLSConfiguration_AllFieldsCombined(t *testing.T) {
+	mdd := &modelDeploymentData{}
+	tlsConfig := &v1alpha2.TLSConfig{
+		VerifyDisabled:  false,
+		CACertSecretRef: "my-ca-bundle",
+		CACertSecretKey: "bundle.crt",
+		UseSystemCAs:    true,
+	}
+
+	addTLSConfiguration(mdd, tlsConfig)
+
+	// Verify all environment variables are set
+	expectedEnvVars := map[string]string{
+		"TLS_VERIFY_DISABLED": "false",
+		"TLS_USE_SYSTEM_CAS":  "true",
+		"TLS_CA_CERT_PATH":    "/etc/ssl/certs/custom/bundle.crt",
+	}
+
+	if len(mdd.EnvVars) != len(expectedEnvVars) {
+		t.Errorf("Expected %d environment variables, got %d", len(expectedEnvVars), len(mdd.EnvVars))
+	}
+
+	for expectedName, expectedValue := range expectedEnvVars {
+		found := false
+		for _, env := range mdd.EnvVars {
+			if env.Name == expectedName {
+				found = true
+				if env.Value != expectedValue {
+					t.Errorf("Expected %s=%s, got %s", expectedName, expectedValue, env.Value)
+				}
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected environment variable %s not found", expectedName)
+		}
+	}
```

Comment:
extreme nit: instead of this O(n^2) setup, we could set up an initial pass to create a map from the `mdd.EnvVars` then compare that final map to the expectedEnvVars. 

it could help simplify the logic as well to something like below
```go
// the failure checks may change if we use testify
actualEnvs := envVarToMapHelper(mdd.EnvVars)
for expectedName, expectedValue := range expectedEnvVars {
	if val, ok := actualEnvs[expectedName]; ok {
		if val != expectedValue {
			t.Errorf() // value does not match
		}
	} else {
		t.Errorf() // environment var DNE
	}
}
```